packages/ember/lib/index.js

```diff
@@ -181,8 +181,10 @@ Object.defineProperty(Ember, 'LOG_VERSION', {
 });
 
 Object.defineProperty(Ember, 'MODEL_FACTORY_INJECTIONS', {
-  get()      { return ENV.MODEL_FACTORY_INJECTIONS;    },
-  set(value) { ENV.MODEL_FACTORY_INJECTIONS = !!value;  },
+  get()      { return false; },
+  set(value) {
+    Ember.logger.warn('As of 2.13.x MODEL_FACTORY_INJECTIONS has no power, feel free to remove it from app/app.js');
```

Comment:
@stefanpenner - Setting it to true has literally no effect in production builds.  The only usage that I can see in Ember Data is [here](https://github.com/emberjs/data/blob/e6cb564bdb33da4b3c15c3c668dc4b1fdafdf190/addon/-debug/index.js#L37) which is _also_ in an `if (DEBUG)` and therefore stripped in production builds.  

The only thing we need to make sure to do here, is blow up if someone attempts to `Ember.ENV.MODEL_FACTORY_INJECTIONS = <anything>` in debug builds, because in production builds there is no behavior changes anyways...